src/primitive.rs

```diff
@@ -0,0 +1,81 @@
+//! Primitives for constructing instances of [`IO`].
+
+use std::{ fmt::{ Display, Debug }, path::Path, fs };
+
+use super::{ IO, Inner::* };
+
+/// An alias for `pure(())`, but with a `'static` lifetime.
+#[allow(non_upper_case_globals)]
+pub const empty: IO<'static> = IO (Pure (()));
```

Comment:
Is this really needed? Shouldn't pure(()) produce a 'static already?

Reply:
yeah the docs are kinda bad here,  It's supposed to convey that it can only be `'static`

Reply:
Updated the comment. Thoughts?